e2e/filesystem-upload.test.js

```diff
@@ -204,4 +206,41 @@ describe('FileSystemUpload end-to-end tests', function() {
         should(events[0].event).be.exactly('fileerror');
     });
 
+    it('GB18030 folder upload test', async function() {
+        const folderName = `aem-upload-e2e_中文_${new Date().getTime()}`;
+        const targetFolder = `${getAemEndpoint()}/content/dam`;
+        const uploadOptions = new DirectBinaryUploadOptions()
+            .withUrl(targetFolder);
+
+        setCredentials(uploadOptions);
+
+        const httpClient = getHttpClient(uploadOptions);
+
+        await createAemFolder(httpClient, uploadOptions, folderName);
+
+        const fileSystemUpload = new FileSystemUpload(getTestOptions());
+
+        monitorEvents(fileSystemUpload);
+
+        const uploadUrl = `${targetFolder}/${encodeURI(folderName)}`;
+        uploadOptions.withUrl(uploadUrl);
+
+        const uploadResult = await fileSystemUpload.upload(uploadOptions, [
+            Path.join(__dirname, 'images/climber-ferrata-la-torre-di-toblin.jpg'),
+        ]);
+
+        should(uploadResult).be.ok();
+
+        await verifyExistsInAemAndHasEvents(httpClient, uploadOptions, '/climber-ferrata-la-torre-di-toblin.jpg');
+
+        const doubleEncodedUrl = `${targetFolder}/${encodeURI(encodeURI(folderName))}`;
+        uploadOptions.withUrl(doubleEncodedUrl);
+
+        const doubleEncodedExists = await doesAemPathExist(httpClient, uploadOptions, '/climber-ferrata-la-torre-di-toblin.jpg');
```

Comment:
Is it OK to use this name `climber-ferrata-la-torre-di-toblin`?

Reply:
The asset came from the asset set that AEM provides in its sample content, so I assume so.